netflix/src/components/home/Preview.tsx

```diff
@@ -0,0 +1,51 @@
+'use client';
+import { useEffect, useState } from 'react';
+import axios from '@/api/axios';
+import request from '@/api/request';
+import Image from 'next/image';
+
+interface Movie {
+  id: number;
+  poster_path: string;
+  title: string;
+}
+
+export default function Preview() {
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  useEffect(() => {
+    // 영화 정보를 불러오는 함수
+    async function fetchMovies() {
+      try {
+        const requests = await axios.get(request.fetchPopular);
+        setMovies(requests.data.results);
+        return requests;
+      } catch (error) {
+        console.error(error);
+      }
+    }
```

Comment:
현재 민영님과 혜연님은 Preview.tsx 컴포넌트의 최상단에 'use client' directive를 써주시고 `useEffect()` 훅을 이용하여 data fetching을 진행하고 계신 것 같아요. 물론 이러한 비동기 요청을 react hook을 통해 진행하는 것도 참 좋지만, nextJS가 풀스택 프레임워크인 만큼, 서버 사이드 렌더링을 적극적으로 활용해봐도 좋을 것 같아요.

react18에서 소개된 "서버 컴포넌트"가 nextJS 앱 라우팅 방식의 핵심이라고 평가받고 있는데, 이 경우에는 해당 컴포넌트 자체를 async 함수로 만들어서 내부적으로 비동기 로직을 펼칠 수 있어요.
```
async function getData() {
  const res = await fetch('https://api.example.com/...')
  // The return value is *not* serialized
  // You can return Date, Map, Set, etc.
 
  if (!res.ok) {
    // This will activate the closest `error.js` Error Boundary
    throw new Error('Failed to fetch data')
  }
 
  return res.json()
}
 
export default async function Page() {
  const data = await getData()
 
  return <main></main>
}
```
이런 식으로요! 이때 fetch() 함수를 적극적으로 활용하면 좋은데, fetch() 함수의 두 번쨰 인자로 cache 옵션을 활용하면 SSR, SSG, ISR 방식 중에 적절한 조절을 할 수 있거든요. 설명 드린 방식을 통하면 `useEffect()` 훅을 쓸 필요도 없고, 따라서 클라이언트 컴포넌트로 선언해줄 필요도 없습니다.

Reply:
이런 코드리뷰..!! 너무 좋습니다